### ISSUE TITLE
Reduced shear correction for DeltaSigma model

### ISSUE THREAD

**Issue by combet** [LSSTDESC/CLMM#537]
When we measure `DeltaSigma` from the data, we basically compute `g*Sigma_c`, where `g` is the reduced shear and not the shear `gamma` (`g = gamma/(1-kappa)`). However, the model definition of `DeltaSigma` is `DeltaSigma = gamma*Sigma_c`.

Far from the halo center, where `kappa<<1`, `g = gamma` and the `DeltaSigma` model (that we have currently in CLMM) that corresponds to `gamma*Sigma_c` is sufficient. However, when `kappa` starts to be "large" (R<~1 Mpc), `kappa` becomes sufficiently large that we need to account for it. In practice, this means multiplying `DeltaSigma` as defined in CLMM's theory modules by
<img width="202" alt="Screenshot 2022-09-22 at 16 26 11" src="https://user-images.githubusercontent.com/20227809/191773848-7a216015-b200-4ebf-9af2-16221a0d5f8b.png">

(See McClintock (2019))

Below the the difference between the two for a 1e15 Msun cluster at z=0.5 and a source at z=2.

<img width="685" alt="Screenshot 2022-09-22 at 16 51 56" src="https://user-images.githubusercontent.com/20227809/191780218-468dde6f-f627-47a1-bda7-2eb739784b5b.png">

We should provide a function that implements the correction.

**Comment by hsinfan1996** [comment]
Does that mean users need to provide z_src to compute Sigma_c in the correction factor?

**Comment by combet** [comment]
Yes, this becomes dependent on the source redshift (either individual or distribution).

**Comment by combet** [comment]
@hsinfan1996, @m-aguena  - I was going through opened issues to see if anything outstanding needed addressing and I came upon this one... I don't think we did anything about it yet, right?

**Comment by hsinfan1996** [comment]
@combet Where is the `DeltaSigma` calculation you are referring to in the code?

**Comment by combet** [comment]
So the DeltaSigma in the theory module corresponds to shear * Sigma_crit. We actually define the shear from DeltaSigma in theory (e.g. parent_class.py, l. 306)
```     
def _eval_tangential_shear_core(self, r_proj, z_cl, z_src):
        delta_sigma = self.eval_excess_surface_density(r_proj, z_cl)
        sigma_c = self.cosmo.eval_sigma_crit(z_cl, z_src)
        return delta_sigma / sigma_c
 ```  
 
But when we compute DeltaSigma from the data, what we do is reduce_shear * Sigma_crit, as the data only gives access to the reduced shear. So in the inner regions, I think the model DeltaSigma will not be the right model for the data and this may biased the mass estimation in a fit. So I think the user should have the option to compute the correction factor to apply to the model before fitting. But then, yes, the model depends on the data through the redshifts...

Not sure if I'm making sense 😅

**Comment by hsinfan1996** [comment]
I see. So basically there should be a version of excess surface density that is corrected by 1/(1-kappa).

**Comment by combet** [comment]
Actually, thinking about it again it may be a non-problem. 

It's really dataops that may be "misleading", as it's multiplying the tan and cross components by sigmacrit when the flag `is_deltasigma=True`. If the tan and cross components are shear (say in simulations where this can be known), then yes, the result will be DeltaSigma and match the definition in theory. If these components are reduced shear (the case in real life), then the result will be not quite DeltaSigma in the inner regions. But maybe it's up to the user to know what they are doing and we can simply expand a bit the docstring of the `is_deltasigma` flag to make that point?

**Comment by hsinfan1996** [comment]
The docstrings for the function `compute_tangential_and_cross_components` in `dataops` mention the concern, but I am not sure if this is enough. We can port this part to the argument `is_deltasigma` and maybe `galaxycluster.py` as well.
https://github.com/LSSTDESC/CLMM/blob/157bdb0ce54d5c98a637be6db69c82f3714a8083/clmm/dataops/__init__.py#L71-L79
In addition, I think the docstrings are not accurate as we now use the `is_deltasigma` flag to control if DeltaSigma profile is the output.

**Comment by combet** [comment]
Ah, thanks, I completely missed the warning. Yes, might be worth moving it to the `is_deltasigma` level in case people don't read further than that.
